### DISCUSSION TITLE
[HELP]"Invalid token"When docker installed the ssp and configured the "Reset Password by email" option

### DISCUSSION THREAD

**Discussion by King-WKR** [ltb-project/self-service-password#1023] [Q&A]
I don't know why i have this issue.
PLEASE help me.
The ldap is 172.26.77.3
SMTP SERVER is another ip: 172.30.3.129
the command is : 
docker run -p 8080:80 \
    -v $PWD/ssp.conf.php:/var/www/conf/config.inc.local.php \
    -it docker.io/ltbproject/self-service-password:latest 

the ssp.conf.php is : 
//<?php // My SSP configuration
$keyphrase = "*****";
$debug = false;
$use_captcha = true;
$ldap_url = "ldap://172.26.77.3:389";
$ldap_binddn = "cn=Man,dn=SHU,dc=local";
$ldap_bindpw = "*****";
$ldap_base = "dc=SHU,dc=local";

$use_tokens = true;
$mail_address_use_ldap = false;
$token_lifetime = "3600";
$mail_attribute = "mail";
$mail_from = "account@SHU.com.cn";
$mail_from_name = "Self Service Password";
$mail_signature = "";
$notify_on_change = true;
$mail_sendmailpath = '/usr/sbin/sendmail';
$mail_protocol = 'smtp';
$mail_smtp_debug = 0;
$mail_debug_format = 'html';
$mail_smtp_host = '172.30.3.129';
$mail_smtp_auth = false;
$mail_smtp_user = '';
$mail_smtp_pass = '';
$mail_smtp_port = 25;
$mail_smtp_timeout = 30;
$mail_smtp_keepalive = false;
$mail_smtp_secure = '';
$mail_smtp_autotls = false;
$mail_smtp_options = array();
$mail_contenttype = 'text/plain';
$mail_wordwrap = 0;
$mail_charset = 'utf-8';
$mail_priority = 3;
//?>

**Comment by davidcoutadeur** [comment]
The invalid token suggests that the token generated by the reset by mail workflow has not been recognized by SSP.

You can check the logs for understanding why this happens.

**Reply by King-WKR** [reply]
[Wed Dec 18 09:02:23.141289 2024] [php:notice] [pid 21:tid 21] [client 172.26.65.134:60732] Invalid form token: sent: 9dce208d72885633199f9acc793227130f27783ff5f0fb05f401a38fbd6cf0e7, stored: , referer: http://172.26.71.249:8080/?action=sendtoken

It seems that token does not work.
should i make some change to php?

**Comment by King-WKR** [comment]
I docker installed the ssp and i received the email.But the website is http://ssp.example.com, in which place i can change this ...
